src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1011,20 +1016,22 @@ private static string GetInputTypeString(InputType inputType)
             }
         }
 
-        private static IEnumerable<SelectListItem> GetSelectListItems([NotNull] ViewContext viewContext, string name)
+        private static IEnumerable<SelectListItem> GetSelectListItems(
+            [NotNull] ViewContext viewContext,
+            string expression)
         {
-            var value = viewContext.ViewData.Eval(name);
+            var value = viewContext.ViewData.Eval(expression);
             if (value == null)
             {
                 throw new InvalidOperationException(Resources.FormatHtmlHelper_MissingSelectData(
-                    "IEnumerable<SelectListItem>", name));
+                    "IEnumerable<SelectListItem>", expression));
```

Comment:
Do a `string.Format` here with `nameof(SelectListItem)`, this way we can have refactoring, F12, all that Jazz support for `SelectListeItem`:

```C#
string.Format("IEnumerable<{0}>", nameof(SelectListItem))
```